python/ccxt/base/exchange.py

```diff
@@ -268,22 +264,15 @@ def define_rest_api(self, api, method_name, options={}):
                     setattr(self, camelcase, partial)
                     setattr(self, underscore, partial)
 
-    def raise_error(self, exception_type, url, method='GET', error=None, details=None):
+    def raise_error(self, exception_type, url=None, method=None, error=None, details=None):
         details = details if details else ''
+        url = url if url else ''
+        method = method if method else ''
         if error:
-            if type(error) is _urllib.HTTPError:
-                details = ' '.join([
-                    str(error.code),
-                    error.msg,
-                    error.read().decode('utf-8'),
-                    details,
-                ])
-            else:
-                details = ' '.join([
-                    str(error),
-                    details,
-                ])
-        raise exception_type(' '.join([self.id, method, url, details]))
+            error = str(error)
```

Comment:
Do we output a HTTP status code here? I'm not sure if str(error) will actually print all of them.
Also, we will have excessive spacing on None-values, this: `' '.join(['', ''])` turns into this: `' '` instead of this: `''` )

Reply:
I agree that the excessive spacing isn't ideal, but the long term goal of this change is to set the arguments to different attributes of the exception class.  Before`str(error)` we had `' '.join([str(error), 'details'])` which did pretty much the same thing.

Reply:
It would be fairly easy to fix this issue - instead of using `' '.join` we could use '{} {} {}'.format(*args)